plugins/search/api-report.md

```diff
@@ -65,10 +66,16 @@ export type FiltersState = {
 //
 // @public (undocumented)
 export const HomePageSearchBar: ({
-  placeholder,
-}: {
-  placeholder?: string | undefined;
-}) => JSX.Element;
+  className: defaultClassName,
+  ...props
+}: Partial<Omit<SearchBarBaseProps, 'onChange' | 'onSubmit'>>) => JSX.Element;
```

Comment:
As you'll see here and below, the destructuring of parameters leaks out into API docs. We should prefer to destructure inside the code instead:

```tsx
export const HomePageSearchBar: (props: Partial<Omit<SearchBarBaseProps, 'onChange' | 'onSubmit'>>) = {
  const { className: defaultClassName, ...otherProps } = props;
  // ...
}
```

or possibly even better, make a props type for this:

```tsx
/**
 * Props for {@link HomePageSearchBar}.
 *
 * @public
 */
export type HomePageSearchBarProps = Partial<Omit<SearchBarBaseProps, 'onChange' | 'onSubmit'>>;

/**
 * Search bar that ... etc etc
 *
 * @public
 */
export const HomePageSearchBar: (props: HomePageSearchBarProps) = {
  const { className: defaultClassName, ...otherProps } = props;
  // ...
}
```

this will maximise the resulting API docs quality and separation.

Same thing applies further down in this file.

EDIT: On further inspection it looks like you DO have a separate type for the props but it gets inlined here by the generator. Whaaaat. Ok disregard that particular part of the comment then :)